lib/plugins/paragon-webpack-plugin/ParagonWebpackPlugin.js

```diff
@@ -0,0 +1,147 @@
+const { Compilation, sources } = require('webpack');
+const fs = require('fs');
+const path = require('path');
+const {
+  getParagonVersion,
+  getParagonThemeCss,
+} = require('../../../config/data/paragonUtils');
+const {
+  injectMetadataIntoDocument,
+  getParagonStylesheetUrls,
+  injectParagonCoreStylesheets,
+  injectParagonThemeVariantStylesheets,
+} = require('./utils');
+const resolvePrivateEnvConfig = require('../../resolvePrivateEnvConfig');
+
+// Allow private/local overrides of env vars from .env.development for config settings
+// that you'd like to persist locally during development, without the risk of checking
+// in temporary modifications to .env.development.
+resolvePrivateEnvConfig('.env.private');
+
+// Resolve configuration `env.config`.
+const envConfigFile = path.resolve(process.cwd(), 'env.config.js');
+
+// Get Paragon and brand versions / CSS files from disk.
+const paragonVersion = getParagonVersion(process.cwd());
+const paragonThemeCss = getParagonThemeCss(process.cwd());
+const brandVersion = getParagonVersion(process.cwd(), { isBrandOverride: true });
+const brandThemeCss = getParagonThemeCss(process.cwd(), { isBrandOverride: true });
+
+/**
+ * 1. Injects `PARAGON_THEME` global variable into the HTML document during the Webpack compilation process.
+ * 2. Injects `<link rel="preload" as="style">` element(s) for the Paragon and brand CSS into the HTML document.
+ */
+class ParagonWebpackPlugin {
+  constructor({ processAssetsHandlers = [] } = {}) {
+    this.pluginName = 'ParagonWebpackPlugin';
+    this.paragonThemeUrlsConfig = {};
+    this.paragonMetadata = {};
+
+    // List of handlers to be executed after processing assets during the Webpack compilation.
+    this.processAssetsHandlers = [
+      this.resolveParagonThemeUrlsFromConfig,
+      this.injectParagonMetadataIntoDocument,
+      this.injectParagonStylesheetsIntoDocument,
+      ...processAssetsHandlers,
+    ].map(handler => handler.bind(this));
+  }
+
+  /**
+   * Resolves the MFE configuration for ``PARAGON_THEME_URLS`` in the following priority order:
+   *
+   * 1. ``envConfig.PARAGON_THEME_URLS``
+   * 2. ``process.env.PARAGON_THEME_URLS``
+   *
+   * @returns {Object} Metadata about the Paragon and brand theme URLs from configuration.
+   */
+  async resolveParagonThemeUrlsFromConfig() {
+    let envConfig;
+    if (fs.existsSync(envConfigFile)) {
+      try {
+        envConfig = require(envConfigFile);
+      } catch (error) {
+        console.info('Paragon Plugin cannot load env.config file, if you want to preload the theme url be sure your file is a CommonJS module');
+      }
+    }
+
+    const paragonThemeUrls = envConfig?.PARAGON_THEME_URLS ?? process.env.PARAGON_THEME_URLS;
```

Comment:
a )  It is treated as a string
b )  I think we should not encourage the use of the string of that variable because would be complicated to maintain. 

Might this one be a better approach?
```suggestion
    const paragonThemeUrls = envConfig?.PARAGON_THEME_URLS ?? {};
```

Reply:
I think your proposed update makes sense given `process.env.PARAGON_THEME_URLS` is parsed as a string. That is, falling back to `{}` here seems reasonable.

We can probably also remove the `resolvePrivateEnvConfig('.env.private');` towards the beginning of this file if the plugin no longer reads from `process.env` env vars 😄

Reply:
Y made the changes, thanks @adamstankiewicz